libraries/core_libs/network/src/tarcap/packets_handlers/common/packet_handler.cpp

```diff
@@ -82,14 +81,9 @@ bool PacketHandler::sealAndSend(const dev::p2p::NodeID& node_id, SubprotocolPack
     return false;
   }
 
-  const auto [peer, is_pending] = peers_state_->getAnyPeer(node_id);
-  if (!peer) [[unlikely]] {
-    LOG(log_wr_) << "sealAndSend failed to find peer";
-    return false;
-  }
-
-  if (is_pending && packet_type != SubprotocolPacketType::StatusPacket) [[unlikely]] {
-    LOG(log_wr_) << "sealAndSend failed initial status check, peer " << node_id.abridged() << " will be disconnected";
+  const auto peer = peers_state_->getPacketSenderPeer(node_id, packet_type);
+  if (!peer.first) {
+    LOG(log_er_) << "Unable to send packet. Reason: " << peer.second;
```

Comment:
yeah probably not. On the other hand it should happen pretty much rarely and it might be nice to see if it is not happening too often, which would indicate some other problem ?